### PR TITLE
add back non-camelCased flags for `skip-install` and `skip-test`

### LINKED ISSUE #274: 1.4.1 ignores --skip-install option
yo@1.4.1 seems to ignore commend options like `--skip-install` or `--skip-test`, but the options without dash in between works normally.

Even if I specify the value like `--skip-install=true` still no luck. By the way, I can use `yo generator MyApp -singleDashOrTwo` in 1.3.3, but that doesn't work unless I use two dash instead, that means neither `yo -v` nor `yo -h` works.

> Env
yo@1.4.1
npm@2.1.18
node v0.10.35

### PR DESCRIPTION
**Pull request by sindresorhus**
fixes #274

### CODE REVIEW

**Comment by SBoudrias** [comment]
For me this is not sufficient.

With 1.4, we break every generator who relied on options being passed as is. either `this.options['skip-install']` or any custom ones `this.options['any-options-really']`. This is problematic as we are introducing a major breaking changes that is affecting all of the yeoman ecosystem without backward compatibility.

**Comment by sindresorhus** [comment]
That wasn't clear in the original issue.

**Comment by sindresorhus** [comment]
See updated commit. Now adds back un-camelized versions of all flags.

I think this is the right way to go. Going forward we'll want to use the camelCased variant.

File: lib/cli.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ var opts = cli.flags;
 var args = cli.input;
 var cmd = args[0];
 
+// add un-camelized options to for legacy
+// TODO: remove some time in the future when generators have upgraded
+Object.keys(opts).forEach(function (key) {
+  opts[key.replace(/[A-Z]/g, function (m) {
```
Comment (SBoudrias) [inline lib/cli.js]: LGTM, I'd just put the replace call out of the square brackets.

Comment (sindresorhus) [inline lib/cli.js]: done

**Pull request merged**